ivy/array/array.py

```diff
@@ -140,11 +140,14 @@ def __init__(self, data, dynamic_backend=None):
     def _init(self, data, dynamic_backend=None):
         if ivy.is_ivy_array(data):
             self._data = data.data
+        elif ivy.is_native_array(data):
```

Comment:
We should instead update the condition checked in `ivy.utils.assertions` so that it returns `True` if `data` is either a `ivy.NativeArray` or a `np.ndarray` rather than adding more conditions here. And finally we could replace `self._data` with `self._data=ivy.asarray(data).to_native()` as you've done in the `elif`

Reply:
I think we must have a condition for `if native: self._data=data` because it will be stuck in a loop of creating another `ivy.Array`?

Reply:
Yep, you're right 👍
Then the only change that would be needed is `ivy.asarray` instead of `ivy.current_backend().asarray` and `._data` for faster access

Reply:
Got it, thank you for the thorough explanation :1st_place_medal: